controllers/contentlibrary/v1alpha2/contentlibraryitem/contentlibraryitem_controller.go

```diff
@@ -245,7 +249,32 @@ func (r *Reconciler) setUpVMIFromCLItem(ctx *context.ContentLibraryItemContextA2
 	vmi.Status.Name = clItem.Status.Name
 	vmi.Status.ProviderItemID = string(clItem.Spec.UUID)
 
-	return nil
+	return addContentLibraryRefToConversionAnnotation(ctx)
+}
+
+// addContentLibraryRefToConversionAnnotation adds the conversion annotation with the content
+// library ref value populated.
+func addContentLibraryRefToConversionAnnotation(ctx *context.ContentLibraryItemContextA2) error {
+	if ctx.CLItem.Status.ContentLibraryRef == nil {
+		return nil
+	}
+
+	current := ctx.VMI.DeepCopyObject().(conversion.Hub)
+
+	prevVMI := &v1alpha1.VirtualMachineImage{}
+	prev := prevVMI.DeepCopyObject().(conversion.Convertible)
```

Comment:
Why are we making a copy of an empty object here?

Reply:
_The entire controller logic is to get the existing v1a2 object, down convert it to v1a1, then set the contentLibraryRef and store this entire object in the conversion annotation._ 

The empty object here is to get an instance of type `Convertible` which then calls `ConvertFrom` resulting in the conversion logic to be exercised thus giving a populated v1a1 object.

Reply:
The v1a1 `VirtualMachineImage` implements `Convertible` though. There's some existing test code that does that on the image: https://github.com/vmware-tanzu/vm-operator/blob/main/api/v1alpha1/virtualmachineimage_conversion_test.go#L37